src/models/vehicle/Vehicle.tsx

```diff
@@ -1,17 +1,17 @@
 import { MathUtils, PerspectiveCamera, Vector3 } from 'three'
-import { useRef, useLayoutEffect, useEffect } from 'react'
+import React, { useRef, useLayoutEffect, useEffect } from 'react'
```

Comment:
This due how typescript is compiled. Yes vite currently handles it great and pass it trough to ESBuild. Typescript's compiler won't: 'React' refers to a UMD global, but the current file is a module. Consider adding an import instead. I'm pretty sure I can fiddle around typescript's config to avoid this problem.

Reply:
hm i think that cant be right. TS definitively supports the transform. it's not just about the global, it has also performance implications. see: https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#jsx-factories